### PR TITLE
MOTECH-2867 Saving null if empty string in field

### PR DESCRIPTION
**Pull request by tomaszzawadzki**

### CODE REVIEW

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

**Comment by motech-gerrit** [comment]
Can one of the admins verify this patch?

File: platform/mds/mds-web/src/main/resources/webapp/js/controllers.js [hunk 1]
```diff
@@ -4196,6 +4196,9 @@
 
             var values = $scope.currentRecord.fields;
             angular.forEach (values, function(value, key) {
+                if(value.type.defaultName === "str" && value.value === ""){
```
Comment (sebbrudzinski) [inline platform/mds/mds-web/src/main/resources/webapp/js/controllers.js]: Why are we extracting the default name parameter from the type for this check? Isn't there any more appropriate field?

Comment (tomaszzawadzki) [inline platform/mds/mds-web/src/main/resources/webapp/js/controllers.js]: defaultname
Data looks like this. If we want to check the type of field if it is string type i think we need this defaultName value.

Comment (sebbrudzinski) [inline platform/mds/mds-web/src/main/resources/webapp/js/controllers.js]: What about typeClass? It clearly indicates a string field.

Comment (tomaszzawadzki) [inline platform/mds/mds-web/src/main/resources/webapp/js/controllers.js]: Yes, typeClass looks better. I will change that.

**Comment by sebbrudzinski** [comment]
test this please

**Pull request merged**